### DISCUSSION TITLE
not sure how to fix this

### DISCUSSION THREAD

**Discussion by Hykiinyte** [HaxeFlixel/flixel#2653] [Q&A] [answered]
This is my first time using Haxe, so I'm doing their little tutorial. I'm a bit stumped on this and don't know how to fix it.
Player.hx down below
(there were originally way more errors, but i'vs managed to fix some)

```hx
package;

import flixel.FlxG;
import flixel.FlxSprite;
import flixel.math.FlxPoint;
import flixel.util.FlxColor;

class Player extends FlxSprite
{
	static inline var SPEED:Float = 200;

	var newAngle(default, null):Float;

	public function new(x:Float = 0, y:Float = 0)
	{
		super(x, y);

		makeGraphic(16, 16, FlxColor.BLUE);
		drag.x = drag.y = 1600;
	}

	// WASD handler
	function updateMovement()
	{
		var up:Bool = false;
		var down:Bool = false;
		var left:Bool = false;
		var right:Bool = false;

		up = FlxG.keys.anyPressed([UP, W]);
		down = FlxG.keys.anyPressed([DOWN, S]);
		left = FlxG.keys.anyPressed([LEFT, A]);
		right = FlxG.keys.anyPressed([RIGHT, D]);

		// cancel opposite keys
		if (up && down)
			up = down = false;
		if (left && right)
			left = right = false;

		// move when key pressed
		if (up || down || left || right)
		{
			var newAngle:Float = 0;
			if (up)
			{
				newAngle = -90;
				if (left)
					newAngle -= 45;
				else if (right)
					newAngle += 45;
			}
			else if (down)
			{
				newAngle = 90;
				if (left)
					newAngle += 45;
				else if (right)
					newAngle -= 45;
			}
			else if (left)
				newAngle = 180;
			else if (right)
				newAngle = 0;
		}

		velocity.set(SPEED, 0);
		velocity.rotate(FlxPoint.weak(0, 0), newAngle);
	}
}
```

**Comment by Geokureli** [comment, marked as answer]
when asking for help, be sure to say what actual problems you're running into. are there compile errors, or runtime errors? If so, list them. is something behaving different than how you expected? if so what behavior did you expect and what are you observing?

Don't just post an entire class and say that some unspecified problem exists, somewhere

**Reply by Hykiinyte** [reply]
Sorry, I'm getting this:

For function argument 'Angle' tasks [Ln 67, Col 40]

**Reply by Geokureli** [reply]
that error is part of another error, if you're on  VSCode it should show all errors, but just to be sure try running the lime build command manually and checking the output. for instance

> rooms/RoomState.hx:119: characters 17-18 : Int should be states.OgmoEntityData<props.DoorValues>
> rooms/RoomState.hx:119: characters 17-18 : ... For function argument 'data'

You can also mouse over the line of code to see all errors on that line, like so
<img width="640" alt="Screen Shot 2022-10-13 at 4 38 47 PM" src="https://user-images.githubusercontent.com/2609513/195716059-4005d5f0-0ec9-4368-9063-a4a58ea528bd.png">
